### PR TITLE
clarify independence of volume lifecycle

### PR DESCRIPTION
**Pull request by trapier**
The fact that containers, services, and stacks can implicitly create data volumes is nice.  It also makes it incumbent upon the operator to understand:

1. If an existing volume name matches one in a container or service spec, then the existing volume options take precedence over those on the container or service specification.  For example, if an existing volume named `mydata` is of type `tmpfs` and a service spec calls for `mydata` with type `local` and options set for nfs, tasks will be deployed using the existing `tmpfs` volume.

2. None of `docker {container,service,stack} rm` remove volumes.  When it comes time to clean up or reclaim storage, it is the operators responsibility.

Signed-off-by: Trapier Marshall <trapier.marshall@docker.com>

### CODE REVIEW

File: engine/swarm/services.md [hunk 1]
```diff
@@ -851,9 +851,14 @@ don't specify a type.
 
 #### Data volumes
 
-Data volumes are storage that remain alive after a container for a task has
-been removed. The preferred method to mount volumes is to leverage an existing
```
[the remark this answers is not in the document]
Comment (trapier) [reply engine/swarm/services.md]: Original recommendation here https://github.com/moby/moby/pull/25197#discussion_r72727323

If there isn't a clear consensus on dropping the recommendation, then I've no objections to keeping it.

[the remark this answers is not in the document]
Comment (trapier) [reply engine/swarm/services.md]: The issue I take with this recommendation is that swarmkit provides no mechanism to create volumes ahead of time, so if you're not using Swarm or a volume plugin that intercepts `volume create` at the manager API and does the right thing for other nodes, then it's an operational pain in the butt.

[the remark this answers is not in the document]
Comment (trapier) [reply engine/swarm/services.md]: Perhaps @cpuguy83, @wsong, or @anusha-ragunathan would be interested in lending an opinion here.

[the remark this answers is not in the document]
Comment (cpuguy83) [reply engine/swarm/services.md]: I agree it doesn't make sense to create ahead of time for most cases.

[the remark this answers is not in the document]
Comment (anusha-ragunathan) [reply engine/swarm/services.md]: +1 on making it explicit that volume management is (creation/removal) separate from service management and that volumes are typically not available at a cluster level, only at the node level.

**Pull request merged**